packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md

```diff
@@ -0,0 +1,324 @@
+- Stylistic rule configurations, for consistent and predictable syntax usage:
+  - **`stylistic`**: Stylistic rules you can drop in without additional configuration.
+  - **`stylistic-type-checked`**: Additional stylistic rules that require type information.
+
+> `recommended-requiring-type-checking` is now an alias for `recommended-type-checked`.
+> The alias will be removed in a future major version.
+
+As of v6, we recommend that projects enable two configs from the above:
+
+- If you are _not_ using typed linting, enable `stylistic` and either `recommended` or `strict`, depending on how intensely you'd like your lint rules to scrutinize your code.
+- If you _are_ using typed linting, enable `stylistic-type-checked` and either `recommended-type-checked` or `strict-type-checked`, depending on how intensely you'd like your lint rules to scrutinize your code.
+
+For example, a typical project that enables typed linting might have an ESLint configuration file that changes like:
+
+```js title=".eslintrc.cjs"
+module.exports = {
+  extends: [
+    'eslint:recommended',
+    // Removed lines start
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    // Removed lines end
+    // Added lines start
+    'plugin:@typescript-eslint/recommended-type-checked',
+    'plugin:@typescript-eslint/stylistic-type-checked',
+    // Added lines end
+  ],
+  plugins: ['@typescript-eslint'],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    // Remove this line
+    project: './tsconfig.json',
+    // Add this line
+    project: true,
+    tsconfigRootDir: __dirname,
+  },
+  root: true,
+};
+```
+
+See [our _Configurations_ linting docs](/linting/configs) for the updated documentation on configurations provided by typescript-eslint.
+
+For more information on these changes, see:
+
+- [Configs: Have recommended/strict configs include lesser configs, and simplify type checked names](https://github.com/typescript-eslint/typescript-eslint/discussions/6019) for the discussion leading up to these configuration changes.
+- [feat(eslint-plugin): rework configs: recommended, strict, stylistic; -type-checked](https://github.com/typescript-eslint/typescript-eslint/pull/5251) for the pull request implementing the changes.
+
+### Updated Configuration Rules
+
+Every new major version of typescript-eslint comes with changes to which rules are enabled in the preset configurations - and with which options.
+Because this release also includes a reworking of the configurations themselves, the list of changes is too large to put in this blog post.
+Instead see the table in [Changes to configurations for 6.0.0](https://github.com/typescript-eslint/typescript-eslint/discussions/6014) and [Configs: Last round of "final" changes to configs for v6](https://github.com/typescript-eslint/typescript-eslint/discussions/7130) for a full description of the changes.
```

Comment:
The discussions are nice because they give everyone full context - but they require a lot of context to understand fully.

It might be worthwhile also just generating a diff for each of the configs.
EG just JSON dump then do a patch diff of the config change for recommended / recommended-type-checked.
Maybe could chuck it in this blog post behind an expando section?

Reply:
Yeah that's a great idea!